### PR TITLE
Refactor version functions

### PR DESCRIPTION
**Pull request by passivestar**
Use generic functions to check for blender and data versions

### CODE REVIEW

File: versioning.py [hunk 1]
```diff
@@ -786,7 +786,7 @@ def update_routine(name):
 
     # Special update for opening Blender 2.7x file
     filepath = get_addon_filepath() + "lib.blend"
-    if is_created_before_280() and is_greater_than_280() and bpy.data.filepath != filepath:
+    if is_created_before(2, 8) and is_bl_newer_than(2, 8) and bpy.data.filepath != filepath:
```
Comment (ucupumar) [inline versioning.py:789]: Don't forget about this one also, it should be `is_created_before(2, 80)`

**Review by ucupumar — CHANGES_REQUESTED** [review]
All function inputs for Blender 2.80 checking need to be corrected

**Comment by passivestar** [comment]
Rebased on master and fixed 2.80, pls test with old blender because I'm not sure if old python supports optional params

File: versioning.py [hunk 2]
```diff
@@ -717,7 +717,7 @@ def update_yp_tree(tree):
     # SECTION III: Updates based on the blender version and yp version
 
     # Version 1.1.0 and Blender 2.90 can hide default normal input
-    if is_greater_than_290() and (is_created_before_290() or 
+    if is_bl_newer_than(2, 9) and (is_created_before(2, 9) or
```
Comment (ucupumar) [inline versioning.py:720]: Also don't forget about this, it's actually Blender 2.90

**Review by ucupumar — CHANGES_REQUESTED** [review]
I'm pretty sure optional parameters are supported in Python 3.4. Also don't forget about Blender 2.90.

**Comment by passivestar** [comment]
`is_created_using()` actually was unused :)

**Comment by passivestar** [comment]
> Anything else need to change before merging?

I just tested in the latest blender and in 2.80 by adding different layers, didn't get any errors in the console. If all looks good to you I think this should be good to go now

**Comment by ucupumar** [comment]
Okay, I'm merging it now. Thanks for the PR! 👍

**Pull request merged**